### ISSUE TITLE
TLS handshake error in AssetLib

### ISSUE THREAD

**Issue by EAinsley** [godotengine/godot#97762]
### Tested versions

- Reproducible in v4.4.dev2.official [97ef3c837], v4.4.dev3.official [f4af8201b] and later. (error -110 and -28928)
- In v4.3.stable.official [77dcf97d8], v4.4.dev1.official [28a72fa43]. It gives different error. (error -9984)

### System information

Godot v4.4.dev2 - Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3060 Ti (NVIDIA; 31.0.15.3623) - AMD Ryzen 5 5600X 6-Core Processor (12 Threads)

### Issue description

When change pages in Asset Store, it will sometimes give TLS handshake error.

![TLS error](https://github.com/user-attachments/assets/6585c487-6179-40db-a167-63d8e7a1757e)

~And sometimes the editor will crash.~ (This is unrelated and have been fixed)

Computer network connection is fine and no proxy used.

Probably related to #79162 .

### Steps to reproduce

- Create an empty project.
- Click AssetLib
- (Change pages)
- wait some time

### Minimal reproduction project (MRP)

N/A

**Comment by JekSun97** [comment]
I tested this on 4.4 dev3 and it does happen
![err](https://github.com/user-attachments/assets/fc28b86d-3b6d-4413-b6e3-cd8d708c8ada)

**Comment by HolonProduction** [comment]
I've only been able to get the -110 handshake error which appears since 8ffb7699af98790b6a39c0d425945145e199e67a (even without changing pages rapidly) (this also was introduced in dev3 and the issue was reported against dev2).

`ImageLoaderSVG: Failed to create SVG from buffer, error code 30.` shouldn't be related, this happens when assets use unsupported image formats (like gif) because the editor will try to parse the data as svg if no other supported format matches.

**Comment by timothyqiu** [comment]
Sounds like #97078. Please try the latest `dev3` and see if it still crashes.

**Comment by EAinsley** [comment]
> Sounds like #97078. Please try the latest `dev3` and see if it still crashes.

You're right. The crash has been fixed, but TLS handshake error still exists. I will update the issue.

**Comment by dugramen** [comment]
I noticed it only happens when the editor setting `asset_library/use_threads` is true. The number of errors also matches the number of icons that didn't load.

**Comment by akien-mga** [comment]
I ran into some of these errors in 4.4.beta3, but they seemed fixed by #102774.

Now in the latest `master` branch I only seem to reproduce this one rarely:
```
ERROR: TLS handshake error: -9984
   at: _do_handshake (modules/mbedtls/stream_peer_mbedtls.cpp:88)
mbedtls error: returned -0x2700
```

In my test it happens on this asset: https://godotengine.org/asset-library/asset/2995
which uses an icon from a HTTPS URL with an invalid certificate.

I suggest we can close this issue as fixed as the handshake errors -110 and -28928 were fixed by #102774. I'll open a new issue for handshake error -9984.
